exonum/src/encoding/serialize/json.rs

```diff
@@ -129,27 +129,29 @@ macro_rules! impl_deserialize_bigint {
     ($($name:ty);*) => ($(impl_deserialize_bigint!{@impl $name})*);
 }
 
-/*
 macro_rules! impl_deserialize_float {
-    (@impl $traitname:ident $typename:ty) => {
-        impl<'a> ExonumJson for $typename {
-            fn deserialize(value: &Value, buffer: &'a mut Vec<u8>,
-                            from: usize, to: usize ) -> bool {
-                    value.as_f64()
-                         .map(|v| v as $typename)
-                         .map(|val| val.write(buffer, from, to))
-                         .is_some()
+    (@impl $typename:ty) => {
+        impl ExonumJson for $typename {
+            fn deserialize_field<B: WriteBufferWrapper>(value: &Value,
```

Comment:
Could you address somewhere how conversion to / from JSON works for `f32` / `f64`? There are values that have a finite decimal representation, which do not exactly map to a `f32` / `f64` value (e.g., `0.1`, because it is an infinite fraction in the binary base), and some can lead to ambiguous results when converting from JSON and back (e.g., `9876543211234567` for `f64` - it has 16 significant digits, which is more than `floor(53 * log_2(10)) = 15` precision provided by the `f64` type).